docs/topics/testing/advanced.txt

```diff
@@ -600,9 +600,13 @@ utility methods in the ``django.test.utils`` module.
 
 .. function:: setup_test_environment()
 
-    Performs any global pre-test setup, such as the installing the
-    instrumentation of the template rendering system and setting up
-    the dummy email outbox.
+    Performs global pre-test setup, like installing instrumentation for the
+    template rendering system and setting up the dummy email outbox.
+
+    This function also saves the values of settings that it changes so that
```

Comment:
I think raising an error is okay. It might affect a few people, but hopefully it's ultimately helping them fix a problem in their code. If we get reports of that causing problems, we can always change it to do a no-op.